packages/smooth_app/lib/background/background_task_folksonomy.dart

```diff
@@ -0,0 +1,198 @@
+import 'package:flutter/material.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/background/background_task.dart';
+import 'package:smooth_app/background/background_task_queue.dart';
+import 'package:smooth_app/background/operation_type.dart';
+import 'package:smooth_app/database/dao_folksonomy.dart';
+import 'package:smooth_app/database/dao_transient_folksonomy.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/l10n/app_localizations.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_operation.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_provider.dart';
+import 'package:smooth_app/query/product_query.dart';
+
+class BackgroundTaskFolksonomy extends BackgroundTask {
+  BackgroundTaskFolksonomy._({
+    required super.processName,
+    required super.uniqueId,
+    required super.stamp,
+    required this.barcode,
+  });
+
+  BackgroundTaskFolksonomy.fromJson(super.json)
+    : barcode = json[_jsonTagBarcode] as String,
+      super.fromJson();
+
+  final String barcode;
+
+  String? _bearerToken;
+  static const String _jsonTagBarcode = 'barcode';
+  static const OperationType _operationType = OperationType.folksonomy;
+
+  static Future<void> addTask(
+    final String barcode,
+    final LocalDatabase localDatabase,
+  ) async {
+    final String uniqueId = await _operationType.getNewKey(
+      localDatabase,
+      barcode: barcode,
+    );
+    final BackgroundTask task = _getNewTask(uniqueId, barcode);
+    await task.addToManager(localDatabase, queue: BackgroundTaskQueue.fast);
+  }
+
+  static BackgroundTask _getNewTask(
+    final String uniqueId,
+    final String barcode,
+  ) => BackgroundTaskFolksonomy._(
+    processName: _operationType.processName,
+    uniqueId: uniqueId,
+    stamp: ';folksonomy;$barcode',
+    barcode: barcode,
+  );
+
+  @override
+  (String, AlignmentGeometry)? getFloatingMessage(
+    AppLocalizations appLocalizations,
+  ) => null;
+
+  @override
+  bool hasImmediateNextTask = false;
+
+  @override
+  Future<void> execute(LocalDatabase localDatabase) async {
+    final DaoTransientFolksonomy daoTransientFolksonomy =
+        DaoTransientFolksonomy(localDatabase);
+    final DaoFolksonomy daoFolksonomy = DaoFolksonomy(localDatabase);
+
+    final FolksonomyOperation? operation = daoTransientFolksonomy
+        .get(barcode)
+        ?.firstOrNull;
```

Comment:
Interesting case: should we use another background task, again and again, or do everything in the same background task?
We may need a specific "folksonomy" queue for that.